torch/csrc/distributed/autograd/context/dist_autograd_container.h

```diff
@@ -33,14 +38,27 @@ class DistAutogradContainer {
   void releaseContext(int64_t context_id);
 
   // Retrieve the autograd context for a given context_id.
-  const DistAutogradContext& retrieveContext(int64_t context_id) const;
```

Comment:
Is removing the second const due to `autograd_context_lock_`? Shall we make the mutex mutable instead?